content/chainguard/chainguard-registry/pull-through-guides/cloudsmith-pull-through/index.md

```diff
@@ -0,0 +1,163 @@
+
+When pulling from a private Chainguard Registry, the **Upstream URL** must be set to `https://cgr.dev`; any other URL here will cause an error.
+
+Lastly, you need to add the username and password you received when you generated the pull token to the upstream source. To do this, expand the **Authentication** section and under **Method** select **Username and Password**. Then enter the username and password you noted down earlier in their respective fields. 
+
+![Screenshot of the Cloudsmith "Create Docker Upstream Source" window. It shows a "Name" field with the value "Chainguard Public Upstream", the "Priority" set to "1", the "Upstream URL" set to "https://cgr.dev/", and the "Cache and Proxy" radial button selected under the "Mode" options. Additionally, the "Authentication" section has been expanded, with the "Username and Password" method selected and values entered into the "Username" and "Password" fields.](cloudsmith-4.png)
+
+Finally, click the **Create Docker Upstream** button. With that, you're ready to test a Chainguard Production Image through Cloudsmith.
+
+
+### Testing pull through of a Chainguard Production image:
+
+As with testing pull through of a Developer image, you'll first need to authenticate to Cloudsmith:
+
+```sh
+docker login docker.cloudsmith.io
+```
+
+After running the command, you will be able to pull any Production Images that your organization has access to through Cloudsmith. To do so, you would run a command with syntax like the following:
+
+```sh
+docker pull docker.cloudsmith.io/<cloudsmith-organization>/<cloudsmith-repository>/<chainguard-registry>/IMAGE
+```
+
+As with the `docker pull` command used in the previous section, you will need to change the `<cloudsmith-organization>` and `<cloudsmith-repository>` values to reflect your own Cloudsmith setup. Additionally, be sure to change `<chainguard-registry>` to the name of your organization's Chainguard registry.
+
+As an example, the following command will pull the `python:3.9-dev` image from a Chainguard registry named `chainguard.edu` and through a Cloudsmith repository named `chainguard-private` owned by a Cloudsmith organization named `chainguard-example`:
+
+```sh
+docker pull docker.cloudsmith.io/chainguard-example/chainguard-private/chainguard.edu/python:3.9-dev
+```
+
+Once this command is completed you will find the Production Image you pulled in your Cloudsmith repository.
+
+![Screenshot of a portion of a Cloudsmith "Repository" page showing three node packages, each labeled "chainguard.edu/node".](cloudsmith-5.png)
+
+If you run into issues pulling images like this, be sure that your `docker pull` command specifies the correct Cloudsmith organization and repository as well as the correct Chainguard registry. 
```

Comment:
I never got the pull to work, got different errors at various points, but I do see the images in cloudsmith.

Reply:
TBH I wasn't getting consistent outputs either, which is why I'm not showing any output in this tutorial. 

It "works" though, so I'm putting it on Cloudsmith to fix.